daskms/optimisation.py

```diff
@@ -111,13 +111,16 @@ def cached_array(array):
     dsk = dict(array.__dask_graph__())
     keys = set(flatten(array.__dask_keys__()))
 
+    if name is None:
+        name = uuid.uuid4().hex
```

Comment:
Might be worth considering adding a prefix string here - just a precaution for eventual debugging. Something like `gencache` perhaps?

Reply:
Misnamed the variable. It should be ``token``. Further changes are in https://github.com/ska-sa/dask-ms/pull/107/commits/164c5420bf1e479c506559f1e0414c547b3d8da9.

Reply:
Also, token exists to uniquely identify the cache during pickling. Exists so that the graph can be serialised and sent to the distributed scheduler.